shared/model/certificate.model.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseEntity, Instant } from './types';
+
+export interface Certificate extends BaseEntity {
+  name: string;
+  description: string;
+  publicKey: string;
+  privateKey: string;
+  certificate: string;
+  expiry: Instant;
+}
+
+export interface CertificateDTO extends BaseEntity {
+  name: string;
+  description: string;
+  publicKey: string;
+  certificate: string;
+  expiry: Instant;
+}
+
+export interface CertificateCommandDTO {
+  name: string;
+  description: string;
+  regenerateCertificate: boolean;
+  options: CertificateOptions | null;
+}
+
+export interface CertificateOptions {
+  commonName: string;
+  countryName: string;
+  stateOrProvinceName: string;
+  localityName: string;
+  organizationName: string;
+  keySize: number;
+  daysBeforeExpiry: number;
```

Comment:
Shouldn't we have a private_key_passphrase field (stored as OibSecret) ?

Reply:
i hesitated but i don't see the use. private key is stored encrypted in the database so the only use case would be to download the private key in a encrypted mode and use a passphrase for that. But at certificate generation it does seem redundant. Do you agree?

Reply:
Indeed it will be useful only when downloading the private key, and to protect it with a passphrase. But we can do it by design by protecting the private key with a passphrase that will be used inside OIBus at run time AND to protect the private key by default when downloading it.

Anyway, it can also be an improvement, but the private_key_passphrase field in the database can then be removed for now, since it's never used

Reply:
ha yeah it was a leftover in the knex migration i intended to remove it, it is done